### PR TITLE
NMS-14566: Features should go in featuresBoot.d (docs)

### PR DESCRIPTION
**Pull request by dino2gnt**
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14566

### CODE REVIEW

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 1]
```diff
@@ -55,43 +55,12 @@ The chapter _Configure Flow Processing_ contains an example on how to automatica
...
-      wrap, \
-      bundle/4.3.2, \
-      config/4.3.2, \
-      kar/4.3.2, \
-      sentinel-jms, \ <1>
-      sentinel-flows <2>
-
-# ....
+echo "sentinel-jms" | sudo tee -a \${SENTINEL_HOME}/etc/featuresBoot.d/features.boot <1>
+echo "sentinel-flows" | sudo tee -a \${SENTINEL_HOME}/etc/featuresBoot.d/sentinel-features.boot <2>
```
Comment (dino2gnt) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:63]: The way that this is rendered sucks, but it's not any worse than what was already there.  Suggestions welcome.

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 2]
```diff
@@ -55,43 +55,12 @@ The chapter _Configure Flow Processing_ contains an example on how to automatica
...
 ----
 <1> Install and Start JMS communication feature
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:65]:
```suggestion
<1> Install and start JMS communication feature.
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 3]
```diff
@@ -55,43 +55,12 @@ The chapter _Configure Flow Processing_ contains an example on how to automatica
...
 <2> Install and Start Sentinel Flows feature
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:66]:
```suggestion
<2> Install and start Sentinel flows feature.
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc (hunk 1 above)
Comment (Bonrob2) [reply docs/modules/reference/pages/configuration/sentinel-features.adoc:63]: Looks ok to me.

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
minor changes. Thanks!

**Comment by mmahacek** [comment]
Is there anything here that couldn't be targetd to `foundation-2022` instead of 30.x?

File: docs/modules/operation/pages/alarm-correlation/situation-feedback.adoc [hunk 4]
```diff
@@ -14,7 +14,11 @@ Enable situation feedback by installing the feature from the Karaf shell:
 
 `feature:install opennms-situation-feedback`
 
-To ensure that the feature continues to be installed on subsequent restarts, add `opennms-situation-feedback` to the `featuresBoot` property in the `$\{OPENNMS_HOME}/etc/org.apache.karaf.features.cfg`.
+To ensure that the feature continues to be installed on subsequent restarts, add `opennms-situation-feedback` to a file in `featuresBoot.d`:
+[source, console]
+----
+echo "opennms-situation-feedback" | sudo tee \${OPENNMS_HOME}/etc/featuresBoot.d/opennms-situation-feedback.boot
```
Comment (mmahacek) [inline docs/modules/operation/pages/alarm-correlation/situation-feedback.adoc:20]: Don't need the `\` when inside a source block
```suggestion
echo "opennms-situation-feedback" | sudo tee ${OPENNMS_HOME}/etc/featuresBoot.d/opennms-situation-feedback.boot
```

File: docs/modules/operation/pages/events/sources/eif-adapter.adoc [hunk 5]
```diff
@@ -27,14 +27,17 @@ opennms> config:property-set port 1828
 opennms> config:update
 ----
 
+To ensure that the feature continues to be installed on subsequent restarts, add `eif-adapter` to a file in `featuresBoot.d`:
+[source, console]
+----
+echo "eif-adapter" | sudo tee \${OPENNMS_HOME}/etc/featuresBoot.d/eif-adapter.boot
```
Comment (mmahacek) [inline docs/modules/operation/pages/events/sources/eif-adapter.adoc:33]:
```suggestion
echo "eif-adapter" | sudo tee ${OPENNMS_HOME}/etc/featuresBoot.d/eif-adapter.boot
```

File: docs/modules/operation/pages/events/sources/kafka.adoc [hunk 6]
```diff
@@ -42,7 +42,11 @@ Next, install the `opennms-kafka-consumer` feature from that same shell:
 feature:install opennms-kafka-consumer
 ----
 
-To ensure that the feature continues to be installed on subsequent restarts, add `opennms-kafka-consumer` to the `featuresBoot` property in the `$\{OPENNMS_HOME}/etc/org.apache.karaf.features.cfg`.
+To ensure that the feature continues to be installed on subsequent restarts, add `opennms-kafka-consumer` to a file in `featuresBoot.d`:
+[source, console]
+----
+echo "opennms-kafka-consumer" | sudo tee \${OPENNMS_HOME}/etc/featuresBoot.d/kafka-consumer.boot
```
Comment (mmahacek) [inline docs/modules/operation/pages/events/sources/kafka.adoc:48]:
```suggestion
echo "opennms-kafka-consumer" | sudo tee ${OPENNMS_HOME}/etc/featuresBoot.d/kafka-consumer.boot
```

File: docs/modules/operation/pages/kafka-producer/enable-kafka.adoc [hunk 7]
```diff
@@ -29,4 +29,8 @@ Next, install the `opennms-kafka-producer` feature from that same shell using:
 feature:install opennms-kafka-producer
 ----
 
-To ensure that the feature continues to be installed on subsequent restarts, add `opennms-kafka-producer` to the `featuresBoot` property in `$\{OPENNMS_HOME}/etc/org.apache.karaf.features.cfg`.
+To ensure that the feature continues to be installed on subsequent restarts, add `opennms-kafka-producer` to a file in `featuresBoot.d`:
+[source, console]
+----
+echo "opennms-kafka-producer" | sudo tee \${OPENNMS_HOME}/etc/featuresBoot.d/kafka-producer.boot
```
Comment (mmahacek) [inline docs/modules/operation/pages/kafka-producer/enable-kafka.adoc:35]:
```suggestion
echo "opennms-kafka-producer" | sudo tee ${OPENNMS_HOME}/etc/featuresBoot.d/kafka-producer.boot
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 8]
```diff
@@ -55,46 +55,15 @@ The chapter _Configure Flow Processing_ contains an example on how to automatica
...
-      wrap, \
-      bundle/4.3.2, \
-      config/4.3.2, \
-      kar/4.3.2, \
-      sentinel-jms, \ <1>
-      sentinel-flows <2>
-
-# ....
+echo "sentinel-jms" | sudo tee -a \${SENTINEL_HOME}/etc/featuresBoot.d/features.boot <1>
+echo "sentinel-flows" | sudo tee -a \${SENTINEL_HOME}/etc/featuresBoot.d/sentinel-features.boot <2>
```
Comment (mmahacek) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:63]:
```suggestion
echo "sentinel-flows" | sudo tee -a ${SENTINEL_HOME}/etc/featuresBoot.d/sentinel-features.boot <2>
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 9]
```diff
@@ -55,46 +55,15 @@ The chapter _Configure Flow Processing_ contains an example on how to automatica
 
 In some cases it might not be sufficient to auto-deploy/configure the container with a `features.xml` file.
 If more flexibility is required it is suggested to modify/copy `*.cfg` and `*.properties` files directly to the `$\{SENTINEL_HOME}/etc` directory.
```
Comment (mmahacek) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:57]: Needs to escape the `*` to not handle as formatting.
```suggestion
If more flexibility is required it is suggested to modify/copy `\*.cfg` and `*.properties` files directly to the `$\{SENTINEL_HOME}/etc` directory.
```

File: docs/modules/reference/pages/telemetryd/protocols/bmp.adoc [hunk 10]
```diff
@@ -187,12 +187,16 @@ If the TimescaleDB extension is not installed, stat tables will not be converted
...
+[source, karaf]
 ----
 feature:install opennms-telemetry-bmp-stats
 ----
 
-To ensure that the feature continues to be installed on subsequent restarts, add `opennms-telemetry-bmp-stats` to the `featuresBoot` property in `$OPENNMS_HOME/etc/org.apache.karaf.features.cfg`.
+To ensure that the feature continues to be installed on subsequent restarts, add `opennms-telemetry-bmp-stats` to a file in `featuresBoot.d`:
+[source, console]
+----
+echo "opennms-telemetry-bmp-stats" | sudo tee \${OPENNMS_HOME}/etc/featuresBoot.d/telemetry-bmp.boot
```
Comment (mmahacek) [inline docs/modules/reference/pages/telemetryd/protocols/bmp.adoc:198]:
```suggestion
echo "opennms-telemetry-bmp-stats" | sudo tee ${OPENNMS_HOME}/etc/featuresBoot.d/telemetry-bmp.boot
```

**Comment by mmahacek** [comment]
While out of scope of this PR, heads up to @Bonrob2 that docs/modules/reference/pages/configuration/sentinel-features.adoc needs some attention.  Lots of formatting inconsistencies.

**Review by emarsh1995 — APPROVED** [review]
Looks all good to me! 😁 Thanks!

**Review by mmahacek — APPROVED** [review]
LGTM

**Review by Bonrob2 — APPROVED** [review]
LGTM!

**Pull request merged**